src/storage/storage_structure/storage_structure_utils.cpp

```diff
@@ -17,6 +17,21 @@ std::pair<BMFileHandle*, page_idx_t> StorageStructureUtils::getFileHandleAndPhys
     }
 }
 
+common::page_idx_t StorageStructureUtils::insertNewPage(BMFileHandle& fileHandle,
```

Comment:
So you are creating this function with a function as an argument but you use it only once. Why bother with the generality and refactoring if there are no other callers? Why not get in a uint8_t* and do memcpy as the default operation and you can extract an insertOp function as an argument if there are other callers.  That said,  are there really no other callers that adds a new page and could use this logic?

Update: I went over the code except for the insertOp, it looks like this function is almost identical to the `StorageStructure::addNewPageToFileHandle()` function, which does not return the newOriginalPage and has a noOp. So you should refactor those (maybe make the insertOp a default operator with a noOp and always return the newPageIdx.

Reply:
Refactored this altogether with `StorageStructure::addNewPageToFileHandle()` and `NodeColumn:: addNewPageToDataFH()`.